src/Conversion/ONNXToKrnl/Math/Reduction.cpp

```diff
@@ -722,6 +734,10 @@ struct ONNXReductionOpLowering : public OpConversionPattern<ONNXReductionOp> {
     SmallVector<IndexExpr, 4> ubs2;
     create.krnlIE.getShapeAsSymbols(input, ubs2);
     Value trueVal = create.math.constant(rewriter.getIntegerType(1), 1);
+    if (enableParallel) {
```

Comment:
I believe that in the generic case, the outermost loop may be a reduction loop, in which case enabling parallelism would not be safe.

Either this is tested at compile time (namely the reduction pattern rules out outermost loop to be a reduction loop), or it is simply disabled in all cases.

The first and last loop nest are fine, just this one.